### ISSUE TITLE
sqlite crash when `--daemon` flag used

### ISSUE THREAD

**Issue by redshark1802** [ElementsProject/lightning#1420]
lightnind crashes after pulling the latest master (89ff46f1e68d6e5eb8a8bfafbb58ae2f367ed082).
The error is ``sqlite3_close: unable to close due to unfinalized statements or unfinished backups``.

This also happens with a completely new sqlite database: I made a backup of the current sqlite db and removed it, started lightningd again with ``--port 0``.

**Comment by redshark1802** [comment]
Yes, this is the command I use to start lightningd:
``./lightningd/lightningd --network bitcoin --log-level debug --alias [hidden] --log-file /home/bitcoin/lightning.log --daemon``

There is no additional information in the log, the last line is just ``Listening on 'lightning-rpc'``

**Comment by ZmnSCPxj** [comment]
It seems some of the recent code rearrangement around db close-fork-db reopen process for `--daemon` failed.

Without `--daemon` I suspect highly it will succeed to start, so if you need to keep your node up then try running without `--daemon` in the meantime.  We will need to investigate this further, please wait.

**Comment by redshark1802** [comment]
Thanks, that works

**Comment by Rav3nPL** [comment]
It still exists, if you start fresh with --daemon.
If you start w/o --daemon and close, then --daemon works fine.

**Comment by 9735dev** [comment]
To follow-up on @Rav3nPL's comment, if you start a new node with the daemon flag, like below, it fails with this error still. You have to start without `--daemon` one time. Then after the first successful run, you can use daemon.

Easy to reproduce with the following as of commit 69cdfba3c85a2ad82e5f587b2e3c9c3349d4a944:
```
/lightning/lightningd/lightningd --daemon --network=regtest --port=9735 --lightning-dir=/tmp/l1 --bitcoin-datadir=/tmp/n1 --log-file=server.log --log-level=debug --bitcoin-cli=/bitcoin/src/bitcoin-cli --bitcoin-rpcuser=n1 --bitcoin-rpcpassword=n1 --override-fee-rates=253/253/253
```
This first start problem appears to be fixed by #1430 commit 6308c1fee276b47b1231d3ecbe1588e6154bf64f.
